openedx/openedx-platform
# [BB-1336] Add a management command to create or update site configuration

~~This PR adds a new custom management command to create or update the site configuration (`SiteConfiguration` model) for LMS and Studio.~~ **Update 2019-11-14**: Since a similar management command was implemented in https://github.com/edx/edx-platform/pull/20919 after this PR was created, this has now been updated to build on top of the changes introduced by that PR.

**Dependencies**: https://github.com/edx/configuration/pull/5504 depends on this PR.

**Screenshots**:  None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: None

**Testing instructions**:

1. Check out the source branch of the PR.
1. Run the management command with the `--help` flag to check its usage - `python manage.py lms create_or_update_site_configuration --help`.
1. Test the management command by passing the value for the `--site-id` argument or the `domain`, only one of which is required.
1. If there is no `Site` instance for the given site ID or the domain, verify that a new `Site` instance is created for it.
1. If there is no `SiteConfiguration` instance for the `Site` corresponding to the given site ID, a new instance is created and the provided values are set in it.
1. The optional site configuration parameters can be provided from a JSON file using the `-f` (or `--filename`) argument, in place of the `--configuration` parameter that is already present.
1. If the optional site configuration parameters are not provided, the site configuration instance is created if it doesn't exist already and left unmodified if it already exists.
1. The site configuration instance can be enabled or disabled by passing either the `--enabled` or `--disabled` flag. Passing one of these flags without passing any site configuration parameters will result in the behavior mentioned in the previous point and in addition will set the `enabled` field to `True` or `False` based on the flag passed.

**Author notes and concerns**:
* This PR was updated on 2019-11-14 to discard some of its changes and implementation to build on top of an existing command.

**Reviewers**
